src/sys/socket/mod.rs

```diff
@@ -197,9 +202,16 @@ use self::ffi::{cmsghdr, msghdr, type_of_cmsg_data, type_of_msg_iovlen, type_of_
 ///
 /// To make room for multiple messages, nest the type parameter with
 /// tuples, e.g.
-/// `let cmsg: CmsgSpace<([RawFd; 3], CmsgSpace<[RawFd; 2]>)> = CmsgSpace::new();`
+///
+/// ```
+/// use std::os::unix::io::RawFd;
+/// use nix::sys::socket::CmsgSpace;
+/// let cmsg: CmsgSpace<([RawFd; 3], CmsgSpace<[RawFd; 2]>)> = CmsgSpace::new();
+/// ```
+#[repr(C)]
 pub struct CmsgSpace<T> {
     _hdr: cmsghdr,
```

Comment:
I think both `_hdr` and `_data` should be changed to just `hdr` and `data`, yeah?

Reply:
Sure.  The code never actually accesses the fields directly either way, so not sure which is clearer.

Reply:
If these are never accessed, then I think this is Rust-y.